history/taskana-simplehistory-provider/src/test/java/acceptance/events/task/CreateHistoryEventOnTaskRerouteAccTest.java

```diff
@@ -0,0 +1,252 @@
+
+    task4 =
+        TaskBuilder.newTask()
+            .classificationSummary(classificationSummary)
+            .workbasketSummary(domainAWorkbasketSummary)
+            .primaryObjRef(DefaultTestEntities.defaultTestObjectReference().build())
+            .buildAndStore(taskService);
+
+    WorkbasketAccessItemBuilder.newWorkbasketAccessItem()
+        .workbasketId(domainAWorkbasketSummary.getId())
+        .accessId("user-1-1")
+        .permission(WorkbasketPermission.OPEN)
+        .permission(WorkbasketPermission.READ)
+        .permission(WorkbasketPermission.APPEND)
+        .buildAndStore(workbasketService);
+
+    WorkbasketAccessItemBuilder.newWorkbasketAccessItem()
+        .workbasketId(domainBWorkbasketSummary.getId())
+        .accessId("user-1-1")
+        .permission(WorkbasketPermission.OPEN)
+        .permission(WorkbasketPermission.READ)
+        .permission(WorkbasketPermission.APPEND)
+        .buildAndStore(workbasketService);
+  }
+
+  @WithAccessId(user = "admin")
+  @Test
+  void should_CreateRerouteHistoryEvent_When_TaskIsRerouted() throws Exception {
+    historyService.deleteHistoryEventsByTaskIds(List.of(task4.getId()));
+    TaskHistoryQueryMapper taskHistoryQueryMapper = getHistoryQueryMapper();
+    List<TaskHistoryEvent> events =
+        taskHistoryQueryMapper.queryHistoryEvents(
+            (TaskHistoryQueryImpl) historyService.createTaskHistoryQuery().taskIdIn(task4.getId()));
+    assertThat(events).isEmpty();
+    taskService.rerouteTask(task4.getId());
+
+    events =
+        taskHistoryQueryMapper.queryHistoryEvents(
+            (TaskHistoryQueryImpl) historyService.createTaskHistoryQuery().taskIdIn(task4.getId()));
+
+    assertThat(events).hasSize(1);
+    String eventType = events.get(0).getEventType();
+    assertThat(eventType).isEqualTo(TaskHistoryEventType.REROUTED.getName());
+    assertRerouteHistoryEvent(
+        events.get(0).getId(),
+        domainAWorkbasketSummary.getId(),
+        domainBWorkbasketSummary.getId(),
+        "admin");
+
+    historyService.deleteHistoryEventsByTaskIds(List.of(task4.getId()));
+  }
+
+  @WithAccessId(user = "admin")
+  @Test
+  void should_CreateRerouteHistoryEvent_When_MultipleTasksAreRerouted() throws Exception {
+    List<String> taskIds = List.of(task1.getId(), task2.getId(), task3.getId());
+    historyService.deleteHistoryEventsByTaskIds(taskIds);
+    TaskHistoryQueryMapper taskHistoryQueryMapper = getHistoryQueryMapper();
+
+    List<TaskHistoryEvent> events =
+        taskHistoryQueryMapper.queryHistoryEvents(
+            (TaskHistoryQueryImpl)
+                historyService.createTaskHistoryQuery().taskIdIn(taskIds.toArray(new String[0])));
+    assertThat(events).isEmpty();
+    taskService.rerouteTasks(taskIds);
+
+    events =
+        taskHistoryQueryMapper.queryHistoryEvents(
+            (TaskHistoryQueryImpl)
+                historyService.createTaskHistoryQuery().taskIdIn(taskIds.toArray(new String[0])));
+
+    assertThat(events)
```

Comment:
Could you rewrite the assertion part?  It would be more readable without the if/else if/else block?

Reply:
How do you suggest I rewrite it?